### DISCUSSION TITLE
Rosetta

### DISCUSSION THREAD

**Discussion by sinan-guler** [wojciech-kulik/xcodebuild.nvim#94] [Q&A] [answered]
Hi, first of all thank you for the fantastic plugin. I would like to ask how I can list, build & run on the Rosetta Simulators ?

**Comment by wojciech-kulik** [comment]
H, thank you! What do you mean by rosetta simulators? You mean to launch the simulator in Rosetta mode?

**Comment by sinan-guler** [comment]
Hi yes in Xcode, you can select the destination architecture and then list the Rosetta variants as well, as shown below. 

<img width="807" alt="SCR-20240317-pkjb" src="https://github.com/wojciech-kulik/xcodebuild.nvim/assets/37443512/e3a34cb3-00aa-49c3-b78b-6c41a1cb1b28">

![image](https://github.com/wojciech-kulik/xcodebuild.nvim/assets/37443512/ee7dbfe3-af24-4259-858b-e39b1d84cb58)

**Reply by wojciech-kulik** [reply]
Does it appear only when you use Intel MacBook?

**Reply by sinan-guler** [reply]
No, I am using Apple Silicon. My previous screenshots were from Xcode 15.2. In Xcode 15.3, they changed the menu. You can achieve the same with: 

Product > Destination > Show All Run Destinations

**Reply by wojciech-kulik** [reply]
Got it now! Does this option simulate running Intel device? 
I will check this

**Comment by wojciech-kulik** [comment, marked as answer]
As much as I was able to research so far, it doesn't seem to be possible using CLI tools. I'm not able to list Rosetta simulators via `xcodebuild`, neither I'm able to check those simulators (their ids) via Xcode.

There are several discussions about this topic and it looks like there is no solution:
https://forums.developer.apple.com/forums/thread/729958
https://github.com/fastlane/fastlane/issues/21293

---

I did some tests and if you run for example `iPhone 15` and `iPhone 15 (Rosetta)` simulators, they both are the same (same IDs) and they both are launched for Apple Silicon (you can check it Activity Monitor). It explains why I can't list them separately.

The only difference is how the app itself is built. In Rosetta mode, you can see in Activity Monitor that the app is running for Intel architecture.

Therefore, you can achieve this behavior by build the app for `x86_64`. To do that you can add an extra build setting in plugin options:
```lua
commands = {
  extra_build_args = "-parallelizeTargets ARCHS=x86_64",
}
```

or you can create a new configuration in your project and switch to it :).

**Reply by sinan-guler** [reply]
Thanks! Still need that for some projects because of old frameworks, I will try these solutions. Thank you for your effort on this subject and for an amazing plugin once again :)
